### PR TITLE
Add ARCO-ERA5 as data source

### PR DESCRIPTION
**Pull request by BSchilperoort**
Due to the discussion on #399 I implemented a loader for the ARCO-ERA5 Zarr dataset (quick Friday afternoon coding exercise).

However, sadly the chunking is set up like:
    `{'time': 1, 'latitude': 721, 'longitude': 1440, 'level': 37}`
Because the data is chunked this way, you cannot download only a spatial subset of the data: the Zarr backend first downloads all the data and only then reduces it. Because of this, it is not very efficient for workflows where you want only a smaller geographical area. Same story for levels.

It does work OK though, but on my home internet (max ~7 MB/s), it takes about a minute to load one time step of data.

## Todo:
- [x] Run test case and compare with normal ERA5 (e.g. the cases in tests/)
- [x] Add user documentation

### CODE REVIEW

**Comment by Peter9192** [comment]
Note that the name of the data is ARCO, not ACOR. 

Would it be worth providing some feedback to them?

**Comment by BSchilperoort** [comment]
> Note that the name of the data is ARCO, not ACOR.

Thanks. I blame covid haha.

> Would it be worth providing some feedback to them?

They are aware of this, see for example [this reply on an issue](https://github.com/google-research/arco-era5/issues/69#issuecomment-2023157545).

They mostly built this for their own use, and made it publicly accessible. I guess that for their use case this chunking makes sense.

**Comment by P-Myers** [comment]
As it seems integration of the online hosted ARCO era5 data into the model is still underway, and the Beta-CDS is still experiencing download delays and minor data instability (mainly for model levels), I was wondering if some additional guidance could be provided on how the ARCO data can be used in the meantime.

Accessing the data is clear, however once downloaded the metadata/ chunking parameters of ARCO data is relatively different than what is expected by WAM2layers, requiring some adjustments. Has a preprocess/adjustment approach been developed for model compatibility? 

Thank you!

Parker

**Comment by BSchilperoort** [comment]
Fixed #425

**Comment by BSchilperoort** [comment]
I also fixed the inconsistent frequency config (where the input frequency could be "1d" while output frequency had to be  "1D")

**Comment by BSchilperoort** [comment]
Longitudes are shifted during preprocessing in this PR. https://github.com/WAM2layers/WAM2layers/issues/425#issuecomment-2545077478 outlines follow-up PR for automagic domain shifting.

**Review by Peter9192 — APPROVED** [review]
Tested with example config eiffel case. Had to

- Change config to use pressure levels. Used those from test workflow
- Add calendar to my config file
- Use a tagging region like [w,s,e,n] as opposed to the mask file that comes with the example case, because of size incompatibility
- Similarly, I set a subdomain for tracking to speed up the calculations
- For visualization, I had to change back to tagging_region mask file, as apparently that doesn't work with a supplied bbox
- For visualization, change frequency to '1D'

After that it seemed to run nicely, and I was able to produce the figure below:

cumulative_sources


Would be nice to have an example config/case for arco-era5 as well. Maybe one of the comparison cases? That could be a new issue/PR though.

File: src/wam2layers/preprocessing/shared.py [hunk 1]
```diff
@@ -265,3 +269,14 @@ def parallel_preprocess(
...
+
+def shift_longitude(ds: xr.Dataset) -> xr.Dataset:
+    """Return dataset with longitude shifted to ISO 6709 standard."""
+    if (
+        (ds["longitude"].to_numpy() > 180).any() or
+        (ds["longitude"].to_numpy() < 0).any()
+    ):
+        ds.coords["longitude"] = (ds.coords["longitude"] + 180) % 360 - 180
+        return ds.sortby(ds.longitude)
+    return ds
```
[the remark this answers is not in the document]
Comment (BSchilperoort) [reply src/wam2layers/preprocessing/shared.py]: This is now logged not as warning but just as info.

File: src/wam2layers/utils/calendar.py [hunk 2]
```diff
@@ -84,7 +84,13 @@ def template_to_files(config: "Config", var: str) -> list[Path]:
 def round_cftime(
     date: CfDateTime, freq: str, how: Literal["nearest", "floor", "ceil"]
 ) -> CfDateTime:
-    date.year, date.month, date.hour, date.minute, date.second, date.calendar, date.has_year_zero
+    """Round a CF datetime index to a nearest frequency."""
+    # make freq argument comply with https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases
+    if any(freq.lower().endswith(_f) for _f in ("h", "s", "min")):
+        freq = freq.lower()
+    else:
+        freq = freq.upper()
```
[the remark this answers is not in the document]
Comment (BSchilperoort) [reply src/wam2layers/utils/calendar.py:92]: that would make this function a lot more complex though. Maybe it's better to first have a discussion somewhere about what people would expect the user frequency inputs to be 🫤

**Pull request merged**